legacy/build-deploy-docker-compose.sh

```diff
@@ -650,13 +650,43 @@ if [[ "$BUILD_TYPE" == "pullrequest"  ||  "$BUILD_TYPE" == "branch" ]]; then
           echo "Using BuildKit for $DOCKERFILE"
       fi
 
+      # Decide whether to use a dedicated project builder for Docker builds - disabled by default.
+      DOCKER_PROJECT_BUILDER=default
+
+      if [ ! -z "$LAGOON_PROJECT_VARIABLES" ]; then
+        DOCKER_PROJECT_BUILDER=($(echo $LAGOON_PROJECT_VARIABLES | jq -r '.[] | select(.scope == "build") | select(.name == "DOCKER_PROJECT_BUILDER") | "\(.value)"'))
+      fi
+      if [ ! -z "$LAGOON_ENVIRONMENT_VARIABLES" ]; then
+        TEMP_DOCKER_PROJECT_BUILDER=($(echo $LAGOON_ENVIRONMENT_VARIABLES | jq -r '.[] | select(.scope == "build") | select(.name == "DOCKER_PROJECT_BUILDER") | "\(.value)"'))
+        if [ ! -z $TEMP_DOCKER_PROJECT_BUILDER ]; then
+          DOCKER_PROJECT_BUILDER=$TEMP_DOCKER_PROJECT_BUILDER
+        fi
+      fi
+
+      case "$DOCKER_PROJECT_BUILDER" in
+        default|Default|False|false|0)
+          DOCKER_PROJECT_BUILDER=default
+          echo "Using default builder for $DOCKERFILE";
+        ;;
+        *)
+          echo "Using dedicated builder $PROJECT for $DOCKERFILE";
+          DOCKER_PROJECT_BUILDER=${PROJECT}
+          docker builder create --name $DOCKER_PROJECT_BUILDER --driver docker-container || true
+        ;;
+      esac
+      set -x
+
```

Comment:
You can condense this entire thing down and use `LAGOON_FEATURE_FLAG_(DEFAULT|FORCE)_DOCKER_PROJECT_BUILDER` rather than some special variable

```
      DOCKER_PROJECT_BUILDER=default
      if [ "$(featureFlag DOCKER_PROJECT_BUILDER)" = enabled ]; then
          DOCKER_PROJECT_BUILDER=${PROJECT}
          docker builder create --name $DOCKER_PROJECT_BUILDER --driver docker-container || true
      fi
```

Reply:
This way someone could enable/disable at the cluster level with either a default `LAGOON_FEATURE_FLAG_DEFAULT_DOCKER_PROJECT_BUILDER` or forcibly with `LAGOON_FEATURE_FLAG_FORCE_DOCKER_PROJECT_BUILDER`

Reply:
Users would set `LAGOON_FEATURE_FLAG_DOCKER_PROJECT_BUILDER=enabled` as a global scope at either project or environment level

Reply:
yah - nicer!